docs/reference/connectors/bigquery/configuration.mdx

```diff
@@ -0,0 +1,383 @@
+              "nullable": false,
+              "description": "Customer ID to filter by"
+            }
+          },
+          "description": "Retrieves customer data by ID"
+        }
+      }
+    },
+    "update_customer_data": {
+      "mutations": {
+        "update_customer_name": {
+          "sql": {
+            "value": "UPDATE `mydataset.customers` SET name = @new_name WHERE customer_id = @customer_id"
+          },
+          "columns": {},
+          "arguments": {
+            "customer_id": {
+              "name": "customer_id",
+              "type": {
+                "scalar_type": "INT64"
+              },
+              "nullable": false,
+              "description": "Customer ID to update"
+            },
+            "new_name": {
+              "name": "new_name",
+              "type": {
+                "scalar_type": "STRING"
+              },
+              "nullable": false,
+              "description": "New customer name"
+            }
+          },
+          "description": "Updates a customer's name"
+        }
+      }
+    }
+  }
+}
+```
+
+#### Native Operation SQL
+
+The SQL for native operations can be specified in two ways:
+
+1. **Inline SQL** (using the `value` property):
+
+   ```json
+   {
+     "sql": {
+       "value": "SELECT * FROM `mydataset.customers` WHERE customer_id = @customer_id"
+     }
+   }
+   ```
+
+2. **SQL from File** (using the `file` property):
+   ```json
+   {
+     "sql": {
+       "file": "queries/get_customer.sql"
+     }
+   }
+   ```
+
+## Updating with Introspection
+
+When your BigQuery schema changes, you'll need to update your data connector configuration to reflect those changes. The
+BigQuery connector provides an introspection process to automatically generate and update the configuration.
+
+The introspection process connects to BigQuery using the provided connection URI and:
+
+1. Discovers accessible datasets (schemas) and their tables/views
+2. Extracts column information, including data types, nullability, and constraints
+3. Identifies primary keys and foreign key relationships where possible
+4. Generates the complete configuration file with all required metadata
+
+To run the introspection process and update your configuration:
+
+1. Ensure you have the correct credentials and permissions to access the BigQuery project
+2. Run the introspection command, providing your connection details
+3. Review the generated configuration file for accuracy
+4. Apply the updated configuration to your data connector
+
+Note that BigQuery has some specific characteristics to consider:
+
+- BigQuery uses datasets as schema equivalents
+- Primary keys are not enforced by BigQuery but can be defined in the configuration
+- Foreign key relationships are not enforced by BigQuery but can be defined in the configuration
+- Certain BigQuery-specific types like GEOGRAPHY, STRUCT, and ARRAY are represented in a way that maintains
+  compatibility with the NDC protocol
```

Comment:
It would be helpful to pull this out as either an `h2` or an admonition.

Reply:
👍